odata-url-conventions/5 Query Options.md

```diff
@@ -93,7 +93,8 @@ to the `null` value using the [`eq`](#Equals) and [`ne`](#NotEquals)
 operators.
 
 When applied to operands of numeric types, [numeric
-promotion](#NumericPromotion) rules are applied.
+promotion](#NumericPromotion) rules are applied. Otherwise primitive
+operands of different types are always unequal.
```

Comment:
This section is for all logical operators, not just equality. Don't know how to express this, maybe we have to add the statement about different types to every section below.

Reply:
The new sentence comes _before_ the `Equals` section, therefore it applies to the other logical operators up to 5.1.1.1.6 as well.

Reply:
Then let's be explicit, because "unequal due to different types" does not help in evaluating "greater than" comparisons. We should explicitly state that this evaluates to false if the types are different and not both numeric.

Reply:
I suggest:
* `'foo' eq 4` and `'foo' in (4)` are false
* `'foo' ne 4` is true
* `'foo' lt 4`, `'foo' le 4`, `'foo' gt 4`, `'foo' ge 4` are invalid

Reply:
"invalid" means "false"?

Reply:
If different instances of `foo` have different types, the property must be `Edm.PrimitiveType` or `Edm.Untyped`, right?